pkg/webhookconfig/common.go

```diff
@@ -203,19 +209,185 @@ func generateValidatingWebhook(name, servicePath string, caData []byte, validati
 			},
 			CABundle: caData,
 		},
-		SideEffects:             &sideEffect,
+		SideEffects:             &noneOnDryRun,
 		AdmissionReviewVersions: []string{"v1beta1"},
 		TimeoutSeconds:          &timeoutSeconds,
 		FailurePolicy:           &failurePolicy,
+		Rules:                   generateRules(rule, operationTypes),
 	}
+}
 
-	if !reflect.DeepEqual(rule, admregapi.Rule{}) {
-		w.Rules = []admregapi.RuleWithOperations{
-			{
-				Operations: operationTypes,
-				Rule:       rule,
-			},
-		}
+func generateObjectMeta(name string, owner ...metav1.OwnerReference) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:            name,
+		OwnerReferences: owner,
+	}
+}
+
+// policy webhook configuration utils
+
+func getPolicyMutatingWebhookConfigName(serverIP string) string {
+	if serverIP != "" {
+		return config.PolicyMutatingWebhookConfigurationDebugName
+	}
+	return config.PolicyMutatingWebhookConfigurationName
+}
+
+func getPolicyValidatingWebhookConfigName(serverIP string) string {
+	if serverIP != "" {
+		return config.PolicyValidatingWebhookConfigurationDebugName
+	}
+	return config.PolicyValidatingWebhookConfigurationName
+}
+
+func constructPolicyValidatingWebhookConfig(caData []byte, timeoutSeconds int32, owner metav1.OwnerReference) *admregapi.ValidatingWebhookConfiguration {
+	name, path := config.PolicyValidatingWebhookName, config.PolicyValidatingWebhookServicePath
+	return &admregapi.ValidatingWebhookConfiguration{
+		ObjectMeta: generateObjectMeta(config.PolicyValidatingWebhookConfigurationName, owner),
+		Webhooks: []admregapi.ValidatingWebhook{
+			generateValidatingWebhook(name, path, caData, timeoutSeconds, policyRule, createUpdate, admregapi.Ignore),
+		},
+	}
+}
+
+func constructDebugPolicyValidatingWebhookConfig(serverIP string, caData []byte, timeoutSeconds int32, owner metav1.OwnerReference) *admregapi.ValidatingWebhookConfiguration {
+	name, url := config.PolicyValidatingWebhookName, fmt.Sprintf("https://%s%s", serverIP, config.PolicyValidatingWebhookServicePath)
+	return &admregapi.ValidatingWebhookConfiguration{
+		ObjectMeta: generateObjectMeta(config.PolicyValidatingWebhookConfigurationDebugName, owner),
```

Comment:
The debug webhook previously didn't have owner refs. Is this an expected change? Do you know why they didn't have owner refs previously?

Reply:
Dunno, shall I preserve the old behaviour ?
It looked better to have owner ref in debug too.